flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java

```diff
@@ -675,7 +676,7 @@ public void close() throws FlinkKafka011Exception {
 						break;
 					case AT_LEAST_ONCE:
 					case NONE:
-						currentTransaction.producer.close();
+						currentTransaction.producer.close(0, TimeUnit.SECONDS);
```

Comment:
Technically speaking, we should only close the producer with timeout zero when it is a exactly once producer. 
A more consistent approach might be always call flush first, then close the producer with 0 timeout. That should work for both exactly-once and at-least-once cases.

Reply:
Good catch! I'll change it to always flush here